### PR TITLE
Get category data from magento

### PR DESCRIPTION
**Pull request by DenysZP**

### CODE REVIEW

File: magento/src/main/java/com/shopapp/magento/api/MagentoApi.kt [hunk 1]
```diff
@@ -107,6 +124,66 @@ class MagentoApi(context: Context) : Api {
...
+            SortType.RECENT -> optionsBuilder.addSortOrder(CREATED_AT_FIELD, true)
+            SortType.PRICE_HIGH_TO_LOW -> optionsBuilder.addSortOrder(PRICE_FIELD, true)
+            SortType.PRICE_LOW_TO_HIGH -> optionsBuilder.addSortOrder(PRICE_FIELD
+            )
+            else -> {
+                /*do nothing*/
+            }
+        }
+
+        if (sortBy == SortType.RECENT) {
```
Comment (radyslavkrechet) [inline magento/src/main/java/com/shopapp/magento/api/MagentoApi.kt]: There is a duplicate of sort order params in options builder.

File: app/src/main/java/com/shopapp/ui/item/CategoryItem.kt [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
-        View.inflate(context, R.layout.item_category, this)
+        val layout = if (isGridMode) R.layout.item_category_grid else R.layout.item_category
+        inflate(context, layout, this)
         setBackgroundResource(R.color.white)
     }
 
     fun setCategory(category: Category) {
-        image.setImageURI(category.image?.src)
-        titleText.text = category.title
+        findViewById<TextView>(R.id.titleText).text = category.title
```
Comment (plzen) [inline app/src/main/java/com/shopapp/ui/item/CategoryItem.kt]: Can we avoid `findViewById` for each category set?

File: app/src/main/java/com/shopapp/ui/item/CategoryItem.kt [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
+import com.shopapp.gateway.entity.Category
 
-class CategoryItem(context: Context) : ConstraintLayout(context) {
+@SuppressLint("ViewConstructor")
+class CategoryItem(context: Context, isGridMode: Boolean = true) : ConstraintLayout(context) {
 
     init {
         layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
-        View.inflate(context, R.layout.item_category, this)
+        val layout = if (isGridMode) R.layout.item_category_grid else R.layout.item_category
```
Comment (plzen) [inline app/src/main/java/com/shopapp/ui/item/CategoryItem.kt]: As for me this is better to achieve with inheritance

File: data/src/main/java/com/shopapp/data/impl/CategoryRepositoryImpl.kt [hunk 4]
```diff
@@ -17,9 +17,9 @@ class CategoryRepositoryImpl(private val api: Api) : CategoryRepository {
         }
     }
 
-    override fun getCategoryList(perPage: Int, paginationValue: String?): Single<List<Category>> {
+    override fun getCategoryList(perPage: Int, paginationValue: String?, rootCategoryId: String?): Single<List<Category>> {
```
Comment (plzen) [inline data/src/main/java/com/shopapp/data/impl/CategoryRepositoryImpl.kt]: Do we really need this for each request? Can we pass this param for `ShopifyApi` class?

Comment (plzen) [reply data/src/main/java/com/shopapp/data/impl/CategoryRepositoryImpl.kt]: Lets rename it to parentCategoryId

**Pull request merged**